<a href="https://colab.research.google.com/github/581aditi/Functions/blob/main/my_first_code.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
# Addition of two numbers
# accepting input from the user:
# input() function is used to accept input from the user
num1= int(input("Enter the first number:"))
#"100" -> 100
num2= int(input("Enter the second number:"))
#"200" -> 200
res =num1 + num2
# 100+200 = 300
print(res) #300

Enter the first number:50
Enter the second number:60
110


In [ ]:
l = [10,-4,23,5,0]
print(l)

[10, -4, 23, 5, 0]


In [ ]:
l[0] = "Aditi"
print(l)

['Aditi', -4, 23, 5, 0]


In [ ]:
s = "Aditi"
print(s)
print(type(s))
s[0] = "F"
print(s)

Aditi
<class 'str'>


TypeError: 'str' object does not support item assignment

In [ ]:
num = -100
if num>0:
   print("Number is positive")
   print("This is inside the if block")
print("This outside if block")

This outside if block


In [ ]:
age = int(input("Enter your age "))
if age >= 18:
  print("You are eligible for driving license")
else:
  print("You are not eligible for driving license")

Enter your age 25
You are eligible for driving license


In [ ]:
age = int(input("Enter your age :"))
if age<=5:
  print("Ticket price : Free")
elif age <=10:
  print("Ticket price : 100")
elif age <=60:
  print("Ticket price : 150")
else:
  print("Ticket price : 100")

Enter your age :25
Ticket price : 150


In [ ]:
marks = int(input("Enter the marks "))
if marks >= 90:
  print("Grade A")
elif marks >= 80:
  print("Grade B")
elif marks >=70:
  print("Grade C")
elif marks >= 50:
  print("Grade D")
else:
  print("Failed !")

Enter the marks :95
Grade A


In [ ]:
name = "Jonathan"
vowels ="aeiouAEIOU"
vowels_count = 0
for char in name:
  if char in vowels:
    vowel_count = vowel_count + 1
    print(vowel_count)

NameError: name 'vowel_count' is not defined

Name : Aditi Sunilkumar Patil
Data Analytics
Github : [link text](https://)

1. Extract **a customer's first name** from a full name using slicing.

In [1]:
name = "Aditi Sunilkumar Patil"
first_name = name[0:5]
print(first_name)

Aditi


 2. Verify if a **given keyword exists** in a product description.

In [8]:
product_name = "Samsung is a high tech smart phone"
keyword = "smart"
if keyword in product_name:
  print("keyword exists")
else:
  print("Keyword doesnot exists")


keyword exists


3. convert a users email adrress into lower case

In [9]:
email = "JOnaTHan00@GMAIL.com"
lower_case = email.lower()
print(lower_case)

jonathan00@gmail.com


4. Replace **a sensitive word** in a financial statement with `"****"`

In [17]:
sensitive_word = ("123456")
print(sensitive_word.replace("123456", "******"))

******


5.Find if a **tracking ID** contains the prefix `"TRK"`.  

In [27]:
a = "TRK12345678"
if( a.startswith("TRK")):
 print("TRK exists")
else:
 print("TRK doesnot exists")

TRK exists


1. Calculate **total salary** after adding a bonus to an employee’s base salary.

In [28]:
total_salary = 50000
bonus = 5000
total_salary = total_salary + bonus
print(total_salary)

55000


2. Compute the **total price** of an order, given unit price and quantity.

In [30]:
unit_price = 250
quantity = 4
total_price = unit_price * quantity
print(total_price)

1000


3. Increase a **bank balance** by adding interest.  

In [31]:
bank_balance = 10000
interest = 500
bank_balance = bank_balance + interest
print(bank_balance)

10500


4. Find the **remaining loan amount** after a partial repayment.  

In [32]:
loan_amount = 75000
repayment = 5000
remaining_loan_amount = loan_amount - repayment
print(remaining_loan_amount)

70000


5. Compute the **total bill** by applying a discount to an initial amount.

In [33]:
initial_bill = 2000
discount = 200
total_bill = initial_bill - discount
print(total_bill)

1800


### **Task 1: String Operations for Business Applications**  
#### **Objective**: Perform industry-relevant string operations.  

#### **Tasks**:
1. Extract **a customer's first name** from a full name using slicing.  
2. Verify if a **given keyword exists** in a product description.  
3. Convert a **user’s email address** to lowercase for uniformity.  
4. Replace **a sensitive word** in a financial statement with `"****"`.  
5. Find if a **tracking ID** contains the prefix `"TRK"`.  

#### **Assume Data**:
- Full Name: `"Alice Johnson"`  
- Product Description: `"This smartphone features a high-resolution display."`  
- Email: `"John.Doe@Company.com"`  
- Sensitive Text: `"Your password is 123456"`  
- Tracking ID: `"TRK987654"`  

---

### **Task 2: Arithmetic and Assignment Operators in Finance**  
#### **Objective**: Perform arithmetic operations for real-world calculations.  

#### **Tasks**:
1. Calculate **total salary** after adding a bonus to an employee’s base salary.  
2. Compute the **total price** of an order, given unit price and quantity.  
3. Increase a **bank balance** by adding interest.  
4. Find the **remaining loan amount** after a partial repayment.  
5. Compute the **total bill** by applying a discount to an initial amount.  

#### **Assume Data**:
- Base Salary: `50000`, Bonus: `5000`  
- Unit Price: `250`, Quantity: `4`  
- Bank Balance: `10000`, Interest: `500`  
- Loan Amount: `75000`, Repayment: `5000`  
- Initial Bill: `2000`, Discount: `200`  

---

### **Task 3: Comparison Operators in Business Applications**  
#### **Objective**: Compare values to validate business conditions.  

#### **Tasks**:
1. Compare **today's stock price** with yesterday’s to check if it increased.  
2. Verify if an **employee's salary** is greater than the company’s average.  
3. Check if a **customer’s age** qualifies for a senior citizen discount.  
4. Determine if a **product’s weight** is within acceptable limits.  
5. Check if a **student's exam score** meets the passing requirement.  

#### **Assume Data**:
- Yesterday’s Stock Price: `150.75`, Today’s Stock Price: `155.20`  
- Employee Salary: `60000`, Average Salary: `55000`  
- Customer Age: `65`, Discount Eligibility Age: `60`  
- Product Weight: `505`, Acceptable Range: `490-510`  
- Exam Score: `75`, Passing Marks: `40`  

---

### **Task 4: Logical Operators in Business Use Cases**  
#### **Objective**: Combine multiple conditions using logical operators.  

#### **Tasks**:
1. Verify if a **loan applicant meets both age and salary requirements**.  
2. Determine if a **warehouse needs restocking** based on low stock or high demand.  
3. Check if an **employee qualifies for overtime pay** (weekend and extra hours).  
4. Confirm if a **transaction is fraudulent** based on large amount or flagged account.  
5. Validate if a **patient requires emergency attention** based on vitals.  

#### **Assume Data**:
- Loan Applicant: Age `30`, Salary `60000`, Required Age `25`, Required Salary `50000`  
- Stock: `15`, Minimum `20`, High Demand: `True`  
- Hours Worked: `10`, Is Weekend: `True`  
- Transaction Amount: `5000`, Average Spend: `1200`  
- Patient Vitals: Heart Rate: `110`, Oxygen Level: `89`, Normal HR: `<100`, Normal O2: `>90`  

---

### **Task 5: Membership Operators in Business Applications**  
#### **Objective**: Check for membership in predefined datasets.  

#### **Tasks**:
1. Check if a **currency code** is valid for a banking system.  
2. Validate if a **user role** exists in a system’s predefined access roles.  
3. Verify if a **flight’s airport code** is recognized in an airline system.  
4. Determine if a **social media post contains a trending hashtag**.  
5. Confirm if a **shipment tracking ID is active**.  

#### **Assume Data**:
- Currency Codes: `"USD INR EUR GBP JPY"`  
- Access Roles: `"Admin Manager Editor Viewer"`  
- Airport Codes: `"LAX JFK DXB HND"`  
- Trending Hashtags: `"#Tech #AI #DataScience"`  
- Active Shipments: `"TRK123 TRK456 TRK789"`  

---
